hazelcast/src/test/java/com/hazelcast/map/MapUnboundReturnValues_StringKeyTest.java

```diff
@@ -0,0 +1,27 @@
+package com.hazelcast.map;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.SlowTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(SlowTest.class)
+public class MapUnboundReturnValues_StringKeyTest extends MapUnboundReturnValuesTestSupport {
```

Comment:
Hmm, that could be hard, since the feature depends heavily on the distribution of data in the cluster. This is why there is a hard-coded minimum of 100k before the feature triggers at all.

What makes the tests so slow is that I check that the feature doesn't trigger below the configured limit.

If we drop that check we could fill a single node with 120k small elements and see if the exception is thrown:
* filling cluster up to a lower bound which should be safe to not trigger
* call `IMap.values()` to check that lower bound didn't trigger
* repeat:
  * fill a bit more data in
  * call `IMap.values()`

If we drop this check we could just fill a single node with 120k items and run a single `Imap.values()` on it to trigger the exception. This should be much faster (single node, just one query).

How long may that test run? I'll give it a try locally.